11.0 mig server environment ir config parameter

TDu opened the pull request.

sbidoul committed:
server_environment_ir_config_parameter

sbidoul committed:
server_environment_ir_config_parameter: work around double ormcache issue

sbidoul committed:
server_environment_ir_config_parameter: handle empty values better

sbidoul committed:
server_environment_ir_config_param: silently replace values by config values

This is much more robust that raising an error, and let modules
load ir.config_parameter from xml data files, while still enforcing
values from the config files.

sbidoul committed:
server_environment_ir_config_parameter: update roadmap

sbidoul committed:
server_environment_ir_config_parameter: migrate to 10.0

rousseldenis committed:
[10.0][FIX] server_environment_ir_config_parameter Avoid loading import

Odoo loads modules that contain static directory, so it wanted to load server_environment_files but it does not exist

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[FIX] missing sudo in server_environment_ir_config_param

Without this sudo get_param would fail when the first user reading a parameter that has changed in the configuration file does not have write access to system parameters.

TDu committed:
[MIG] server_environment_ir_config_parameter: Migraton to 11.0

yajo commented on server_environment_ir_config_parameter/README.rst:
```diff
@@ -0,0 +1,85 @@
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/149/10.0
```
11.0

TDu replied:
Done

yajo reviewed (changes requested):
Just on little detail.

TDu committed:
fixup! [MIG] server_environment_ir_config_parameter: Migraton to 11.0

yajo commented:
Bots are ❌

TDu commented:
About the red test, seems that the package python-mysqldb [in travis.yml](https://github.com/OCA/server-tools/blob/ea9b0d58fcadc7792a9b4f570b32a947c7d5ffcf/.travis.yml#L16) does not support Python 3

yajo commented:
Then please take the chance to remove it.

TDu committed:
Start fixing test for Python 3

TDu commented:
Waiting for server_environment [to be migrated](https://github.com/OCA/server-tools/pull/1051)

grindtildeath reviewed (approved):
Tested with https://github.com/OCA/server-tools/pull/1051 and works well :+1:

simahawk commented:
@TDu can you rebase and add the oca_dependencies.txt to include #1051 until is merged?

simahawk reviewed (changes requested):
LGTM, just include #1051 to make travis happy :)

TDu committed:
Add oca_dependencies.txt

TDu commented:
I added oca_dependencies.txt
But server_environment depends on server_environment_files. I changed .travis.yml to create it like on branch 10.0 but still not working...

pedrobaeza commented:
Closing in favor of https://github.com/OCA/server-env/pull/8
